tools/cabana/chartswidget.cc

```diff
@@ -230,7 +232,7 @@ void ChartsWidget::setColumnCount(int n) {
 void ChartsWidget::updateLayout() {
   int n = column_count;
   for (; n > 1; --n) {
-    if ((n * (CHART_MIN_WIDTH + charts_layout->spacing())) < rect().width()) break;
+    if ((n * CHART_MIN_WIDTH) < rect().width()) break;
```

Comment:
charts_layout->spacing() returns `-1` for some reason. Might need a different way to compute the padding between the charts so nothing falls of.

Reply:
![image](https://user-images.githubusercontent.com/1314752/215335576-ed96c91f-8e5c-408c-b0e7-2cc670fa9c7c.png)

Reply:
yes,let's move it for now.

Reply:
it returns -1 if use default system setting. we need call setSpacing to set it to a value.

Reply:
Think there is more spacing that we need to consider. I can make more pixels disappear from the right than the spacing between plots is wide.

Reply:
looks like we need setspacing in constructor. Otherwise the chart will be out of bounds：

![Screenshot from 2023-01-29 23-14-39](https://user-images.githubusercontent.com/27770/215336035-b485c08e-9fd8-4232-a2e8-062200d0c210.png)

Reply:
This is with `setSpacing(10)` called. Much better, but not yet perfect.

![image](https://user-images.githubusercontent.com/1314752/215336396-b00833e6-9a06-44e3-b8b9-c228cd4a4162.png)

Reply:
the correct width should be (n*CHART_MIN_WIDTH) + (n - 1) * charts_layout->spacing()

Reply:
> This is with `setSpacing(10)` called. Much better, but not yet perfect.
> 
> ![image](https://user-images.githubusercontent.com/1314752/215336396-b00833e6-9a06-44e3-b8b9-c228cd4a4162.png)

not sure what we missed

Reply:
Looks like I need to use `charts_layout->geometry().width()` to compare to. But that value is not valid from the resize event.

Reply:
I'll just update the visibility in `updateLayout` since that's where the decision about the number of columns is made anyway. Looks like that fixes everything.

Reply:
great!